### ISSUE TITLE
[BUG] REDQ Loss Shape Mismatch

### ISSUE THREAD

**Issue by Jex-y** [pytorch/rl#1939]
## Describe the bug

I am attempting to use the REDQLoss class, however this throws an error when calculating the policy module loss.

## To Reproduce

```python
import torch
from tensordict.nn import TensorDictModule
from tensordict.nn.distributions import NormalParamExtractor
from torch import nn
from torch.distributions import Normal
from torchrl import envs
from torchrl.collectors import SyncDataCollector
from torchrl.data.replay_buffers import ReplayBuffer, samplers, storages
from torchrl.modules import ProbabilisticActor, ValueOperator
from torchrl.objectives import REDQLoss

env = envs.GymEnv('BipedalWalker-v3')

obs_dim = env.observation_spec["observation"].shape[0]
act_dim = env.action_spec.shape[0]

policy_network = nn.Sequential(
    nn.Linear(obs_dim, 64),
    nn.ReLU(),
    nn.Linear(64, 64),
    nn.ReLU(),
    nn.Linear(64, act_dim*2),
    NormalParamExtractor()
)
policy_module = TensorDictModule(policy_network, in_keys=["observation"], out_keys=["loc", "scale"])
policy_module = ProbabilisticActor(
    policy_module, 
    in_keys=["loc", "scale"], 
    spec=env.action_spec,
    distribution_class=Normal,
    return_log_prob=True, # Not sure about this
)

class QValueNetwork(nn.Module):
    def __init__(self, obs_dim, act_dim):
        super().__init__()
        self.obs_dim = obs_dim
        self.act_dim = act_dim
        self.qvalue_network = nn.Sequential(
            nn.Linear(obs_dim + act_dim, 64),
            nn.ReLU(),
            nn.Linear(64, 64),
            nn.ReLU(),
            nn.Linear(64, 1)
        )

    def forward(self, observation, action):
        x = torch.cat([observation, action], dim=-1)
        return self.qvalue_network(x)

qvalue_module = QValueNetwork(obs_dim, act_dim)
qvalue_module = ValueOperator(qvalue_module, in_keys=["observation", "action"])

loss = REDQLoss(
    policy_module,
    qvalue_module,
    num_qvalue_nets=10,
    sub_sample_len=2,
    action_spec=env.action_spec,
)

optimizer = torch.optim.Adam(loss.parameters(), lr=1e-3)

data_collector = SyncDataCollector(
    env, 
    policy_module,
    frames_per_batch=1000,
    total_frames=10_000,
)

replay_buffer = ReplayBuffer(
    storage=storages.LazyTensorStorage(max_size=100_000),
    sampler=samplers.SamplerWithoutReplacement(),
)

update_to_data_ratio = 20
batch_size = 64

tensordict_data = next(iter(data_collector))
replay_buffer.extend(tensordict_data.view(-1).cpu())
for _ in range(update_to_data_ratio):
    batch = replay_buffer.sample(batch_size)
    loss_dict = loss(batch)
    optimizer.zero_grad()
    total_loss = sum(v for k, v in loss_dict.items() if k.startswith("loss_"))
    total_loss.backward()
    optimizer.step()
```

```bash
[/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/torchrl/data/replay_buffers/samplers.py:37](https://file+.vscode-resource.vscode-cdn.net/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/torchrl/data/replay_buffers/samplers.py:37): UserWarning: Failed to import torchrl C++ binaries. Some modules (eg, prioritized replay buffers) may not work with your installation. If you installed TorchRL from PyPI, please report the bug on TorchRL github. If you installed TorchRL locally and/or in development mode, check that you have all the required compiling packages.
  warnings.warn(EXTENSION_WARNING)

[/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/gymnasium/core.py:311](https://file+.vscode-resource.vscode-cdn.net/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/gymnasium/core.py:311): UserWarning: WARN: env.num_envs to get variables from other wrappers is deprecated and will be removed in v1.0, to get this variable you can do `env.unwrapped.num_envs` for environment variables or `env.get_wrapper_attr('num_envs')` that will search the reminding wrappers.
  logger.warn(
[/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/gymnasium/core.py:311](https://file+.vscode-resource.vscode-cdn.net/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/gymnasium/core.py:311): UserWarning: WARN: env.reward_space to get variables from other wrappers is deprecated and will be removed in v1.0, to get this variable you can do `env.unwrapped.reward_space` for environment variables or `env.get_wrapper_attr('reward_space')` that will search the reminding wrappers.
  logger.warn(
[/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/torchrl/objectives/common.py:28](https://file+.vscode-resource.vscode-cdn.net/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/torchrl/objectives/common.py:28): UserWarning: No target network updater has been associated with this loss module, but target parameters have been found. While this is supported, it is expected that the target network updates will be manually performed. You can deactivate this warning by turning the RL_WARNINGS env variable to False.
  warnings.warn(
[/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/torchrl/objectives/common.py:311](https://file+.vscode-resource.vscode-cdn.net/home/ejex/micromamba/envs/RL/lib/python3.10/site-packages/torchrl/objectives/common.py:311): UserWarning: No target network updater has been associated with this loss module, but target parameters have been found. While this is supported, it is expected that the target network updates will be manually performed. You can deactivate this warning by turning the RL_WARNINGS env variable to False.
  warnings.warn(
state_action_value.shape torch.Size([22, 64])
state_action_value_actor.shape torch.Size([10, 64])
action_log_prob_actor.shape torch.Size([64, 4])
```

Shapes recorded using the following code:
```python
# torchrl/objectives/redq.py:507
print("state_action_value.shape", state_action_value.shape)
(
    state_action_value_actor,
    next_state_action_value_qvalue,
    state_action_value_qvalue,
) = state_action_value.split(
    [self.num_qvalue_nets, self.sub_sample_len, self.num_qvalue_nets],
    dim=0,
)
sample_log_prob = tensordict_actor.get(
    self.tensor_keys.sample_log_prob
).squeeze(-1)
(
    action_log_prob_actor,
    next_action_log_prob_qvalue,
) = sample_log_prob.unbind(0)

print("state_action_value_actor.shape", state_action_value_actor.shape)
print("action_log_prob_actor.shape", action_log_prob_actor.shape)
```

## Expected behavior
REDQLoss.forward either validates input shapes and throws an appropriate error explaining the issues or does not throw an error.

## System info
Installed through pip. Python 

```python
import torchrl, numpy, sys
print(torchrl.__version__, numpy.__version__, sys.version, sys.platform)
```
```bash
0.3.0 1.26.4 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:36:39) [GCC 12.3.0] linux
```

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by Jex-y** [comment]
This issue does not occur when using `torchrl.modules.distributions.TanhNormal` instead of `torch.distributions.Normal`. Not sure if this is intended behavior or not.

**Comment by vmoens** [comment]
Could be related to this

https://github.com/pytorch/rl/issues/1933

If using an Independent distribution solves your problem it means we should find a way to make Normal and TanhNormal and other models more interoperable...

**Comment by vmoens** [comment]
Rewriting the distribution definition with
```python
    distribution_class=lambda *args, **kwargs: Independent(Normal(*args, **kwargs), 1),
```
solves the bug

Happy to hear suggestions on how to make this more obvious that the distribution should have an `Independent` wrapper!
